src/lib/operators/delete.cpp

```diff
@@ -70,7 +70,11 @@ void Delete::_finish_commit() {
 
   const auto table_statistics = _table->table_statistics();
   if (table_statistics) {
-    table_statistics->increment_invalid_row_count(num_rows_deleted);
+    // Hackyhacky const cast. We're not changing the table, technically, just its statistics.
+    // This updates the TableStatistics row count by creating and setting new TableStatistics in the Table.
+    std::const_pointer_cast<Table>(_table)->set_table_statistics(std::make_shared<TableStatistics>(
```

Comment:
This new approach seems quite expensive. Creating a new statistics object every time we invalidate a row will kill us in TPC-C. What is the advantage of this?

Reply:
1. Won't kill us, all it does is copy the column statistics shared_ptrs over and decrease the row count.

2. The whole update-statistics-on-every-delete/insert approach is very hacky anyway. Doesn't update the column_statistics at all. 

3. Just replacing the TableStatistics object gets rid of the race condition we'd have if we'd update the row count of the shared statistics object.

Reply:
If it was up to me I'd remove this mechanism entirely but it wasn't the point of this PR to "break" things.